### PR TITLE
completed training tasks

### PR DESCRIPTION
**Pull request by alnasir7**
- Had fewer integration tests since a lot of the edge cases are automatically caught. For example, by JSON object for null inputs.

### CODE REVIEW

File: src/main/User/Services/GetUserInfoService.java [hunk 1]
```diff
@@ -7,13 +7,16 @@
...
+import java.util.Optional;
+
 import org.json.JSONObject;
 import org.slf4j.Logger;
 
 public class GetUserInfoService implements Service {
   private UserDao userDao;
   private Logger logger;
   private String username;
+  private Optional<User> userOption;
```
Comment (crchong1) [inline src/main/User/Services/GetUserInfoService.java:19]: I wouldn't create a new userOption field here, and instead just use the User user underneath.

File: src/main/User/Services/GetUserInfoService.java [hunk 2]
```diff
@@ -25,10 +28,15 @@ public GetUserInfoService(UserDao userDao, Logger logger, String username) {
   @Override
   public Message executeAndGetResponse() {
     if (!ValidationUtils.isValidUsername(this.username)) {
+      return UserMessage.INVALID_PARAMETER;
+    }
+    if (!userDao.get(this.username).isEmpty()) {
```
Comment (crchong1) [inline src/main/User/Services/GetUserInfoService.java:33]: You are calling userDao.get(this.username) many times. Think about how you can just call the dao once.

File: src/main/User/Services/LoginService.java [hunk 3]
```diff
@@ -9,12 +9,14 @@
...
 
+import java.util.Optional;
+
 public class LoginService implements Service {
   private Logger logger;
   private UserDao userDao;
   private String username;
   private String password;
-  private User user;
+  private Optional<User> userOption;
```
Comment (crchong1) [inline src/main/User/Services/LoginService.java:19]: you dont need an optional here, just use a User

**Review by crchong1 — COMMENTED** [review]
lgtm- just a few comments
